runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -155,7 +155,17 @@ bool TR_EscapeAnalysis::isImmutableObject(TR::Node *node)
    if (node->getOpCodeValue() != TR::New)
       return false;
 
-   if (!strncmp("java/lang/Integer", getClassName(node->getFirstChild()), 17))
+   char *className = getClassName(node->getFirstChild());
+
+   if (!strncmp("java/lang/", className, 10) &&
+          (!strcmp("Integer", &className[10]) ||
+           !strcmp("Long", &className[10]) ||
+           !strcmp("Short", &className[10]) ||
```

Comment:
Is there really not a better way? Not saying we should fix this here but this looks like a really bad way to looking for particular classes. I know for recognized methods we have things like `TR::java_util_HashMap_all` to capture all HashMap methods. Wondering if something similar exists for classes.

Reply:
@fjeremic Filip, I'm trying to figure out whether there's a cleaner way of doing this check.

Reply:
@fjeremic Filip, I don't see any existing mechanism for doing what you're suggesting.  Would you like me to open an issue to track that as an improvement?

Reply:
Yes, let's open up an issue. I see similar types of `strcmp` code in other parts of the code base. I do think we need some notion on "recognized classes" to avoid this type of cruft. Thanks!

Reply:
Opened issue #3804